lib/IRGen/IRGenSIL.cpp

```diff
@@ -1941,12 +1922,13 @@ llvm::Value *IRGenSILFunction::createArrayAndSize(
 /// FIXME: This is incomplete. We should derive this from op metadata. We can
 /// get it from the "number_attr" field of the input's "input_arg". The raw op
 /// generator is probably the right place to put this.
-static const char *inputListNumberAttr(StringRef opName) {
-  if (opName == "Pack")
+static const char *inputListNumberAttr(StringRef opName, unsigned inputIdx) {
```

Comment:
I added a comment explaining in more detail what this attr is for and what inputIdx does.

In the process, I discovered that I was handling "Concat" wrong. It's input list has `inputIdx == 1`. So I fixed that.

Since this is like a database of ops, I think it makes most sense to structure it as a big if-elseif statement that figures out what op it is, and then check the `inputIdx` inside each body to decide what to return.

Reply:
Looks great!